core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala

```diff
@@ -536,6 +537,11 @@ class GroupCoordinatorTest {
     // Make sure the NewMemberTimeout is not still in effect, and the member is not kicked
     assertEquals(1, group.size)
 
+    // prepare the mock replica manager again since the delayed join is going to complete
+    EasyMock.reset(replicaManager)
+    EasyMock.expect(replicaManager.getMagic(EasyMock.anyObject())).andReturn(Some(RecordBatch.MAGIC_VALUE_V1)).anyTimes()
```

Comment:
Hmm, why do we need to mock this since replicaManager.getMagic() is only called through replicaManager.handleWriteTxnMarkersRequest()?

Reply:
```GroupMetadataManager#storeGroup``` (https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala#L245) also call ```ReplicaManager.getMagic```.

There are delayed ops are completed by ```timer.advanceClock``` so we have to mock the ```replicaManager.getMagic```. the mock is same to https://github.com/apache/kafka/blob/trunk/core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala#L3823.